cmd/state-installer/cmd.go

```diff
@@ -234,6 +235,12 @@ func execute(out output.Outputer, cfg *config.Instance, an analytics.Dispatcher,
 		}
 	}
 
+	if condition.OnCI() {
+		if _, err := storage.InstallSource(); err == nil {
+			return errs.New("Cannot run state-installer from an installation directory. Run it from an installer instead.")
+		}
+	}
```

Comment:
This has multiple issues:

- You're ignoring the actual error returned and making assumptions about what it means.
- InstallSource's job isn't to assert whether we are running from an install directory, that's a connotation you're only creating here, making this very error prone wrt future iterations of InstallSource.
- This type of assertion should happen in tests, not in the code we ship.